src/core/qgspointlocator.cpp

```diff
@@ -466,8 +466,10 @@ static QgsPointLocator::MatchList _geometrySegmentsInRect( QgsGeometry *geom, co
   for ( auto part = straightGeom.const_parts_begin(); part != straightGeom.const_parts_end(); ++part )
   {
     // Checking for invalid linestrings
-    // A linestring should/(must?) have at least two points
-    if ( qgsgeometry_cast<QgsLineString *>( *part )->numPoints() < 2 )
+    // A linestring should/(must?) have at least two points.
```

Comment:
Must, if we want to comply with the ogc standard, cf [our discussions](https://github.com/qgis/QGIS/pull/9379#discussion_r269432326)

BTW, I wanted (and I forgot...) to add a method to return if the linestring is valid or not. Something like

```
bool QgsCurve::isValid() {
..
  return curve->numPoints() < 2
}
```

I think this code deserves to be in the class QgsCurve / QgsLineString and the method called here.

Reply:
Sure.